js/util/mapbox.js

```diff
@@ -2,78 +2,102 @@
 
 var config = require('./config');
 var browser = require('./browser');
+var url = require('url');
+var querystring = require('querystring');
 
-function normalizeURL(url, pathPrefix, accessToken) {
+function normalizeURL(mapboxURL, pathPrefix, accessToken) {
     accessToken = accessToken || config.ACCESS_TOKEN;
 
     if (!accessToken && config.REQUIRE_ACCESS_TOKEN) {
         throw new Error('An API access token is required to use Mapbox GL. ' +
             'See https://www.mapbox.com/developers/api/#access-tokens');
     }
 
-    url = url.replace(/^mapbox:\/\//, config.API_URL + pathPrefix);
-    url += url.indexOf('?') !== -1 ? '&access_token=' : '?access_token=';
+    var httpsURL = mapboxURL.replace(/^mapbox:\/\//, config.API_URL + pathPrefix);
+    httpsURL += httpsURL.indexOf('?') !== -1 ? '&access_token=' : '?access_token=';
 
     if (config.REQUIRE_ACCESS_TOKEN) {
         if (accessToken[0] === 's') {
             throw new Error('Use a public access token (pk.*) with Mapbox GL JS, not a secret access token (sk.*). ' +
                 'See https://www.mapbox.com/developers/api/#access-tokens');
         }
 
-        url += accessToken;
+        httpsURL += accessToken;
     }
 
-    return url;
+    return httpsURL;
 }
 
-module.exports.normalizeStyleURL = function(url, accessToken) {
-    if (!url.match(/^mapbox:\/\/styles\//))
-        return url;
+module.exports.normalizeStyleURL = function(styleURL, accessToken) {
+    if (!styleURL.match(/^mapbox:\/\/styles\//))
+        return styleURL;
 
-    var split = url.split('/');
-    var user = split[3];
-    var style = split[4];
-    var draft = split[5] ? '/draft' : '';
-    return normalizeURL('mapbox://' + user + '/' + style + draft, '/styles/v1/', accessToken);
+    var urlObject = url.parse(styleURL, true);
+    var formattedQuery = '';
+
+    if (Object.keys(urlObject.query).length !== 0) {
+        formattedQuery = '?' + querystring.stringify(urlObject.query);
+    }
+
+    var parsedURL = url.format(urlObject.protocol + '/' + urlObject.pathname + formattedQuery);
```

Comment:
What's your intent using `url.format` here? As best I can tell, this method is supposed to take a single `Object` argument.